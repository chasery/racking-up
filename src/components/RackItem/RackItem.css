.RackItem {
  display: flex;
  align-items: center;
}
.RackItem__name {
  flex-grow: 1;
  padding: 0.75rem 0;
}
