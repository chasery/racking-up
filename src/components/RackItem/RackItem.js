import React from "react";
import { useParams } from "react-router-dom";
import { currencyFormat } from "../../currencyFormat";
import "./RackItem.css";

function RackItem(props) {
  const { rackId } = useParams();
  const { name, url, price } = props;
  const formattedPrice = currencyFormat.format(price);

  if (name && url && price) {
    return (
      <li className="RackItem">
        <a
          className="RackItem__info"
          href={url}
          target="_blank"
          rel="noreferrer"
        >
          <span className="RackItem__infoName">{name}</span>
          {price && (
            <span className="RackItem__infoPrice">{formattedPrice}</span>
          )}
        </a>
        {rackId && (
          <div className="RackItem__controls">
            <button>E</button>
            <button>D</button>
          </div>
        )}
      </li>
    );
  } else {
    return (
      <li className="RackItem">
        <span className="RackItem__infoName">
          This rack has no clothing items.
        </span>
      </li>
    );
  }
}

export default RackItem;
