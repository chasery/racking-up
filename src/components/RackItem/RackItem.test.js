import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RackItem from "./RackItem";

describe("RackItem component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<RackItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
