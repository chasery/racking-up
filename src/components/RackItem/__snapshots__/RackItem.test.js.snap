// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RackItem component renders without error 1`] = `
<li
  className="RackItem"
>
  <span
    className="RackItem__infoName"
  >
    This rack has no items.
  </span>
</li>
`;
